### PR TITLE
feat(client): Enable communication prefs for all known English locales.

### LINKED ISSUE #3266: Show communication preferences for all en-* dialects
Users whose browser reports en-GB, en-CA or other non-en-US english dialects will not see the "communication preferences" option in the settings page, and I assume will not see the "get the latest news and tips" opt-in either.  Should we just open this up to en-* or do we have plans to one-day provide translated content for these other locales?

I'll ask the engagement team to weigh in on this one.

(Am I opening a geo-political can of worms here?)

### PR DESCRIPTION
**Pull request by shane-tomlinson**
List from http://www.science.co.il/language/locale-codes.asp

fixes mozilla/fxa-content-server#3266

@vladikoff or @rfk  - r?

### CODE REVIEW

File: communication-prefs/communication-prefs.js [hunk 1]
```diff
@@ -12,7 +12,19 @@ module.exports = {
...
+      'en-cb',
+      'en-gb',
+      'en-ie',
+      'en-in',
+      'en-jm',
+      'en-nz',
+      'en-ph',
+      'en-tt',
       'en-us',
+      'en-za',
```
Comment (vladikoff) [inline communication-prefs/communication-prefs.js]: are we missing en-zw ?

image

Comment (vladikoff) [inline communication-prefs/communication-prefs.js]: image

**Comment by shane-tomlinson** [comment]
Maybe better is just to allow `en`, then use a RegExp for `en-*`

**Comment by vladikoff** [comment]
> Maybe better is just to allow en, then use a RegExp for en-\* 

Yessss and regular `en`. :+1:

**Comment by shane-tomlinson** [comment]
@vladikoff - ready for re-review!

**Comment by TDA** [comment]
Travis seems to be coughing up on the quotes @shane-tomlinson

**Comment by shane-tomlinson** [comment]
Thanks @TDA - I had to disable the quotes check for that line.

File: communication-prefs/communication-prefs.js [hunk 2]
```diff
@@ -5,14 +5,14 @@ module.exports = {
   startDate: '2015-01-01',
   subjectAttributes: ['lang'],
   independentVariables: ['communicationPrefsVisible'],
-  eligibilityFunction: function () {
-    return true;
+  eligibilityFunction: function (subject) {
+    return !! (subject && subject.lang);
```
Comment (shane-tomlinson) [inline communication-prefs/communication-prefs.js:9]: @vladikoff - updated to add the check!

**Comment by vladikoff** [comment]
Thank you Shane!

**Pull request merged**